packages/core/frontend/sirius-components-core/src/representationmetadata/useRepresentationMetadata.tsx

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { gql, useLazyQuery } from '@apollo/client';
+import { useMultiToast } from '../toast/MultiToast';
+import { RepresentationMetadata } from '../workbench/Workbench.types';
+import {
+  GQLRepresentationMetadata,
+  GQLRepresentationMetadataQueryData,
+  GQLRepresentationMetadataQueryVariables,
+  UseRepresentationMetadataValue,
+} from './useRepresentationMetadata.types';
+
+const getRepresentationMetadata = gql`
+  query getRepresentationMetadata($editingContextId: ID!, $representationId: ID!) {
+    viewer {
+      editingContext(editingContextId: $editingContextId) {
+        representation(representationId: $representationId) {
+          id
+          label
+          kind
+        }
+      }
+    }
+  }
+`;
+
+const notNullNorUndefined = (
+  metadata: GQLRepresentationMetadata | null | undefined
+): metadata is GQLRepresentationMetadata => !!metadata;
+
+export const useRepresentationMetadata = (): UseRepresentationMetadataValue => {
```

Comment:
Why all this complexity instead of adding a simple `EditingContextRepresentationsDatafetcher`?

Reply:
Because I don't want to retrieve all representation metadata every time someone update the selection.
I want to try to retrieve the representation metadata of selected elements.

Reply:
Simplified